stdconv/string.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "libmodmqttconv/converter.hpp"
+
+class StringConverter : public DataConverter {
+    public:
+        virtual MqttValue toMqtt(const ModbusRegisters& data) const {
+            std::vector<uint16_t> registers = data.values();
+            ConverterTools::adaptToNetworkByteOrder(registers);
+            size_t size = std::min(mMaxSize, data.getCount() * sizeof(uint16_t));
+            return MqttValue::fromBinary(registers.data(), size);
+        }
+
+        virtual ModbusRegisters toModbus(const MqttValue& value, int registerCount) const {
+            const uint16_t* source = static_cast<const uint16_t*>(value.getBinaryPtr());
+            const int valueCount = value.getBinarySize() / sizeof(uint16_t);
```

Comment:
I do not quite get the idea of max size argument. If we have 9 registers as in your real example we can write:
 - 18 bytes if device does require c-style nul terminated string data
 - 17 bytes if device want us to end string with nul terminated char.

max size is defined by register count. I would just write max 18 bytes or less if string is shorter.

Your implementation allows to write a substring, but only from the beginning. I do not think that this is usable at the converter level.

Reply:
... to clarify I would assume that 0x0 terminating character is always needed when writing. So if you have 18 bytes to write you can only write 17 and end with 0x0.

Reply:
The `mMaxsize` arg has not effect on writing (`toModbus()`) but only on reading (`toMqtt()`), as described in my previous comment.

`toModbus()` takes the mininum of `MqttValue.getBinarySize()` (actual content size) and `registerCount` (function argument). This is not meant as a "feature" to write a substring, it just synchronizes the 2 different size requirements.

I don't understand why the converter should add an extra terminating `0x00` byte. When the device requires a trailing `0x00` byte, it should be part of the `MqttValue`.

Reply:
By the way, I don't need `StringConverter.toModbus()` at all because my energy meter has no writable string registers. I just implemented it for the sake of completeness. If you think that this implementation could cause confusion, we can drop it.

Reply:
> I don't understand why the converter should add an extra terminating 0x00 byte. When the device requires a trailing 0x00 byte, it should be part of the MqttValue.

MqttValue hold a binary payload with its length just as std::string. If will not terminate string data:

```
[zork@serenity modbusmqttgw]$ mosquitto_pub -t 'test' -m HELLO

[zork@serenity modbusmqttgw]$ mosquitto_sub -t '#' -F '%X'
48454C4C4F
```


If device has 7 registers (14 bytes) of space for a string and you want to write some HELLO into it, device has to know where is the end of data. A common practice is to terminate with 0x0 value. 

There is no support for writing less bytes that requested `registerCount` in m2mgateway now, so the simplest solution is to write HELLO and fill rest of space with 0x0 values.

Reply:
OK, I changed the implementation to respect `registerCount` exactly.